libcswsscommon/include/capi/dbconnector.h

```diff
@@ -0,0 +1,36 @@
+#ifndef _C_DBCONNECTOR_H
+#define _C_DBCONNECTOR_H
+
+#include <hiredis/hiredis.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void *db_connector_t;
+
+// DBConnector::DBConnector(int db, std::string hostname, int port, unsigned int timeout)
+db_connector_t db_connector_new(int db, const char *hostname, int port, unsigned int timeout);
```

Comment:
>db_connector_new [](http://example.com/codeflow?start=15&length=16)

Did you try SWIG?
[http://zacg.github.io/blog/2013/06/06/calling-c-plus-plus-code-from-go-with-swig/](http://zacg.github.io/blog/2013/06/06/calling-c-plus-plus-code-from-go-with-swig/)